shell/browser/api/electron_api_session.cc

```diff
@@ -426,6 +426,30 @@ v8::Local<v8::Promise> Session::ResolveProxy(gin::Arguments* args) {
   return handle;
 }
 
+v8::Local<v8::Promise> Session::ResolveHost(gin::Arguments* args) {
+  v8::Isolate* isolate = args->isolate();
+  gin_helper::Promise<std::vector<std::string>> promise(isolate);
+  v8::Local<v8::Promise> handle = promise.GetHandle();
+
+  std::string host;
+  args->GetNext(&host);
```

Comment:
this will automatically throw an error if `host` isn't supplied or isn't a string:

```suggestion
v8::Local<v8::Promise> Session::ResolveHost(std::string host) {
  v8::Isolate* isolate = args->isolate();
  gin_helper::Promise<std::vector<std::string>> promise(isolate);
  v8::Local<v8::Promise> handle = promise.GetHandle();
```